### ISSUE TITLE
QueryGranularity incompatible behavior w.r.t timezone in 0.10.0 

### ISSUE THREAD

**Issue by nishantmonu51** [apache/incubator-druid#4218]
In 0.9.x queryGranularity when represented as string was defaulting to use DurationGranularity. Now after the changes in https://github.com/druid-io/druid/pull/3850, it is now defaulted to use PeriodGranularity instead. This change is incompatible w.r.t timezone as both implementations behave differently based on the user timezone. 

Sample code -  
```
    DateTimeZone.setDefault(DateTimeZone.forOffsetHours(5));
    PeriodGranularity periodGranularity = new PeriodGranularity(new Period("P2D"), null, null);
    DurationGranularity durationGranularity = new DurationGranularity(86400000, 0);
    System.out.println(jsonMapper.writeValueAsString(periodGranularity));
    System.out.println(jsonMapper.writeValueAsString(durationGranularity));
```

output - 
```
{"type":"period","period":"P2D","timeZone":"UTC","origin":null}
{"type":"duration","duration":86400000,"origin":"1970-01-01T05:00:00.000+05:00"}
```

Notice the origin in both the versions, PeriodGranularity defaults to UTC timezone if no timezone is specified explicitly while DurationGranularity defaults to the user timezone.

**Comment by nishantmonu51** [comment]
I think we need to call it out in the release notes and make both Duration and Period granularity behave in a consistent way. @himanshug: @gianm: @cheddar: Any thoughts on what should be the expected behavior ?

**Comment by gianm** [comment]
I think this isn't a real bug, because running in non-UTC timezones was never supported. And in UTC, the behavior didn't change.

**Comment by gianm** [comment]
See also configuration.md in the Druid docs which says,

> `-Duser.timezone=UTC` This sets the default timezone of the JVM to UTC. We always set this and do not test with other default timezones, so local timezones might work, but they also might uncover weird and interesting bugs. To issue queries in a non-UTC timezone, see [query granularities](../querying/granularities.html#period-granularities)

And really, I think the new behavior is helping get Druid in a state where we _could_ officially support running in different user timezones. It implemented part of the ask from #2619 and #1204.

**Comment by nishantmonu51** [comment]
the doc mentions following about DurationGranularities - 
"They also support specifying an optional origin, which defines where to start counting time buckets from (defaults to 1970-01-01T00:00:00Z)." 
If I create a DurationGranularity and serialize the query on client side, it will pick the origin based on the timezone of client VM instead of defaulting to 1970-01-01T00:00:00Z. 
So I think, we need to fix the DurationGranularity to be able to distinguish between when a user has actually set origin or not and default to UTC when user has not set any origin. This will make both Period and Duration granularities behave in same way. does that seem a reasonable fix ?

**Comment by gianm** [comment]
Yeah, changing DurationGranularity to set default origins in UTC does seem reasonable.

**Comment by xvrl** [comment]
@gianm this is a bug, while Druid was always required to be **run** in UTC, PeriodGranularity was designed to allow requesting arbitrary timezones at query time and the results returned should reflect the proper UTC offsets in the timestamp. I think this is a duplicate of #4276

**Comment by gianm** [comment]
I thought this issue was just saying that the interpretation of "null" time zone changed from local time to UTC, when the server is running in local time. Which still doesn't sound like a bug to me, since before and after 0.10.0, the interpretation of "null" time zone is UTC if the server is running in UTC (which it should be). So the behavior didn't change in that case.

#4276 sounds like a different issue to me (although maybe the same underlying cause) and that one is clearly a bug.

**Comment by xvrl** [comment]
@gianm ah, I see, in that case I agree with you. I wrote DurationGranularity as something that was implicitly UTC based, so if we now care about the timezone it spits out, we should explicitly set it UTC, but it was never meant to be timezone aware. The assumption when it was written was that the user timezone was already UTC in the first place. Only PeriodGranularity was designed to allow specifying timezones.

**Comment by gianm** [comment]
@xvrl I think your original design is still good. The only change I'd make, if we ever decided to support running in non-UTC time zones, is documenting what the behavior is when the user timezone is not UTC.

**Comment by jihoonson** [comment]
Looks like this was fixed in #4611. I'm closing this issue.
